__tests__/html/assets/activityGrouping.js

```diff
@@ -0,0 +1,705 @@
+  un: 'userNub',
+  ut: 'userOnTop',
+  w: 'wide',
+  wd: 'hide'
+};
+
+function generateURL(state) {
+  const params = {};
+
+  Object.entries(URL_QUERY_MAPPING).forEach(([short, long]) => {
+    const value = state[long];
+
+    // Do not set "wd=0" for easier copy and paste.
+    if (short === 'wd' && !value) {
+      return;
+    }
+
+    if (typeof value !== 'undefined') {
+      params[short] = value === true ? '1' : value === false ? '0' : value + '';
+    }
+  });
+
+  return '#' + new URLSearchParams(params).toString();
+}
+
+const DEFAULT_STATE = {
+  attachmentLayout: false,
+  botAvatarInitials: true,
+  botNub: true,
+  botOnTop: true,
+  hide: false,
+  rtl: false,
+  showAvatarInGroup: 'status',
+  transcriptName: 'simple-messages.json',
+  userAvatarInitials: true,
+  userNub: true,
+  userOnTop: true,
+  wide: false
+};
+
+function getInitialState(defaultValues = {}) {
+  const initialState = {};
+  const params = parseURLParams(location.hash);
+
+  Object.entries(URL_QUERY_MAPPING).forEach(([short, long]) => {
+    const value = params[short];
+
+    if (typeof value === 'undefined') {
+      initialState[long] = defaultValues[long];
+    } else if (value === '1') {
+      initialState[long] = true;
+    } else if (value === '0') {
+      initialState[long] = false;
+    } else {
+      initialState[long] = value;
+    }
+  });
+
+  return initialState;
+}
+
+const ActivityGroupingBadges = () => {
+  const {
+    attachmentLayout,
+    botAvatarInitials,
+    botNub,
+    botOnTop,
+    rtl,
+    showAvatarInGroup,
+    transcriptName,
+    userAvatarInitials,
+    userNub,
+    userOnTop,
+    wide
+  } = useContext(window.WebChatTest.ActivityGroupingContext);
+
+  const badges = useMemo(() => {
+    const badges = [];
+
+    if (attachmentLayout === 'stacked') {
+      badges.push('layout:stacked');
+    } else if (attachmentLayout === 'carousel') {
+      badges.push('layout:carousel');
+    }
+
+    botAvatarInitials && badges.push('bot:initials');
+    botNub && badges.push('bot:nub');
+    botOnTop && badges.push('bot:on-top');
+    rtl && badges.push('view:rtl');
+
+    if (showAvatarInGroup === 'sender') {
+      badges.push('avatar-group:sender');
+    } else if (showAvatarInGroup === true) {
+      badges.push('avatar-group:every');
+    }
+
+    badges.push(`transcript:${transcriptName}`);
+
+    userAvatarInitials && badges.push('user:initials');
+    userNub && badges.push('user:nub');
+    userOnTop && badges.push('user:on-top');
+
+    wide && badges.push('view:wide');
+
+    return badges.sort();
+  }, [
+    attachmentLayout,
+    botAvatarInitials,
+    botNub,
+    botOnTop,
+    rtl,
+    showAvatarInGroup,
+    transcriptName,
+    userAvatarInitials,
+    userNub,
+    userOnTop,
+    wide
+  ]);
+
+  return (
+    <div className="badges" dir="ltr">
+      {badges.map(badge => (
+        <React.Fragment key={badge}>
+          <nobr className="badges__badge">{badge}</nobr>
+          {'\u200B'}
```

Comment:
The `<nobr>` won't break into another line if the HTML is `<nobr>ABC</nobr><nobr>DEF</nobr>`.

But if we have zero-width space in-between, `<nobr>ABC</nobr>&#x200B;<nobr>DEF</nobr>`, it will break into another line.

Adding comment.